### PR TITLE
Allow deleting a template which has databases made from it

### LINKED ISSUE #59: deleting a used template results in inconsistency
Trying to delete a template that has had a database created from it results in the backing database being deleted but the record of the template not.

```
$ test-db template create foo E8EB40EEAE5D11E4AFD54C5C95573682
CREATED foo

$ test-db database create --based-on foo
CREATED 6A837E12B06D11E4AFD54C5C95573682 at apipe-test-db.gsc.wustl.edu:5432

nnutter@apipe-test-db:/$ psql -U test_db -h localhost test_db -c '\l' | grep foo; echo "--------" && psql -U test_db -h localhost test_db -c 'SELECT * FROM database_template WHERE name = $$foo$$;'
 foo                              | genome   | UTF8     | en_US.UTF-8 | en_US.UTF-8 | 
--------
 template_id | name | owner  | note |        create_time         |       last_used_time       
-------------+------+--------+------+----------------------------+----------------------------
          14 | foo  | genome |      | 2015-02-09 09:07:36.533065 | 2015-02-09 09:07:51.287899
(1 row)

$ test-db delete template foo                                                                                                                                                                                                                
Got error response 400: Bad Request at /usr/bin/test-db-delete line 53

$ test-db template list | grep foo
foo

nnutter@apipe-test-db:/$ psql -U test_db -h localhost test_db -c '\l' | grep foo; echo "--------" && psql -U test_db -h localhost test_db -c 'SELECT * FROM database_template WHERE name = $$foo$$;'
--------
 template_id | name | owner  | note |        create_time         |       last_used_time       
-------------+------+--------+------+----------------------------+----------------------------
          14 | foo  | genome |      | 2015-02-09 09:07:36.533065 | 2015-02-09 09:07:51.287899
(1 row)
```

Creating a template and deleting it works fine:

```
$ test-db template create foo E8EB40EEAE5D11E4AFD54C5C95573682                                                                                                                    
CREATED foo

$ test-db delete template foo
DELETED

$ test-db template list | grep foo
```

### PR DESCRIPTION
**Pull request by brummett**
When deleting a template, find all the live database records which reference
this template, and set their template_id to NULL.

This fixes #59

### CODE REVIEW

File: lib/TestDbServer/Command/DeleteTemplate.pm [hunk 1]
```diff
@@ -12,6 +12,15 @@ sub _entity_find_method { 'find_template' }
 sub _entity_id_method { 'template_id' }
 sub _not_found_exception { 'Exception::TemplateNotFound' }
 
+sub execute {
+    my $self = shift;
+
+    my $databases_with_this_template = $self->schema->search_database(template_id => $self->template_id);
```
Comment (nnutter) [inline lib/TestDbServer/Command/DeleteTemplate.pm]: At this point it hasn't validated that the `template_id` is valid but that doesn't look like it causes any problems.

File: lib/TestDbServer/Command/DeleteTemplate.pm [hunk 2]
```diff
@@ -12,6 +12,15 @@ sub _entity_find_method { 'find_template' }
...
+    $databases_with_this_template->update({template_id => undef});
```
Comment (nnutter) [inline lib/TestDbServer/Command/DeleteTemplate.pm]: It's cool that you can just call `update()` on the whole `ResultSet`.

**Comment by nnutter** [comment]
:+1: once tests are passing due to #70.

File: lib/TestDbServer/Command/DeleteTemplate.pm (hunk 1 above)
Comment (brummett) [inline lib/TestDbServer/Command/DeleteTemplate.pm]: It doesn't have to.  It's perfectly fine to have a resultset with 0 rows.  In that case, the next ```update()``` statement would complete successfully and update 0 rows.

**Pull request merged**